wandb/sdk/launch/_project_spec.py

```diff
@@ -76,10 +75,13 @@ def __init__(
         self.target_entity = target_entity
         self.target_project = target_project.lower()
         self.name = name  # TODO: replace with run_id
+        resource_args_build = resource_args.get(resource, {}).pop("build", {})
```

Comment:
wait how can you `.pop` from `{}`

Reply:
wow that's sick, the default prevents a ValueError

Reply:
I've added a comment explaining the reasoning here